tasks/pdp/task_prove.go

```diff
@@ -631,6 +636,76 @@ func (p *ProveTask) getSenderAddress(ctx context.Context) (common.Address, error
 	return address, nil
 }
 
+func (p *ProveTask) cleanupDeletedRoots(ctx context.Context, proofSetID int64, pdpVerifier *contract.PDPVerifier) error {
+
+	removals, err := pdpVerifier.GetScheduledRemovals(nil, big.NewInt(proofSetID))
```

Comment:
Presuming this doesn't need to wait for the message to land?

Reply:
Yeah I think it is ok in this case since we consult chain itself upon proving to check which pieces are ready for removal.